### DISCUSSION TITLE
Max-width modifier for media query (Ex: `-sm:absolute`)

### DISCUSSION THREAD

**Discussion by mrleblanc101** [tailwindlabs/tailwindcss#10263] [Help] [answered]
Hi, I have a suggestion.
So tailwindcss is mobile-first, which mean I put my mobile style, then desktop style to cancel it `absolute md:static`.
For a very complex mobile menu, this is annoying, is there a way to target only below sm (0 to 640px) ?
I'd love if we could add a modifier to media query like we already can for negative value. 
We have `-mr-10`, why not `-sm:absolute` to target bellow `640px` so I dont have to worry about canceling the style on desktop.

**Comment by adamwathan** [comment, marked as answer]
Hey! We have this already, try `max-sm:absolute` 👍🏻

**Reply by mrleblanc101** [reply]
Wow, you're fast ! I just saw #7645 which is basically the same question. My bad !

**Reply by mrleblanc101** [reply]
Wait, I'll need to check up on that 🤔 
Why is it `@media not all and (min-width: 640px) { ... }` instead of `@media (max-width: 639px) { ... }`.
I think that's why I didn't notice.

**Reply by adamwathan** [reply]
> Why is it @media not all and (min-width: 640px) { ... } instead of @media (max-width: 639px) { ... }.

Just because you can use units other than `px` for media queries, and when you start using those units it's not as simple as just "subtract 1". For example if you have a `40rem` media query, the max-width version would have to be something like `39.999rem` which is just kinda silly.

Using `@media not all and (min-width: 40rem)` although a little unintuitive achieves the same thing without any arbitrary math stuff 👍

**Reply by mrleblanc101** [reply]
Thanks for the explaination. 

Maybe the section title should be more explicit that this is the equivalent of a `max-width` media query.

Maybe your explaination should be a note in the doc so you don't have the same question again.
